File: decorator/en_us/python/rectangle.py
from polygon_decorator import PolygonDecorator
"""
    Calculates perimeter like an Polygon, but your way
    to calculate area is especial.
"""
class Rectangle(PolygonDecorator):

    def calculate_area(self):
        self.area = (self.sides[0] * self.sides[1])
        print("Area = " + str(self.area))

File: decorator/en_us/java/Rectangle.java
/**
 * File: Rectangle
 * Role: Concrete Decorator
 * Purpose: Defines the Rectangle particularities.
 * Concrete Decorator class override any Component {@see Geometric} method(s)
 * whose behavior needs to be modified. In this case it is calculateArea().
 */

public class Rectangle extends PolygonDecorator {

    public Rectangle(float[] sides) {
        super(sides);
    }

    @Override
    public void calculateArea() {

        float sides[] = this.getSides();
        float area = sides[0] * sides[1];

        System.out.println("Area = " + area);
    }
}
